contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol

```diff
@@ -0,0 +1,611 @@
+  }
+
+  /// @notice Gets the token address on the remote chain.
+  /// @param remoteChainSelector Remote chain selector.
+  /// @dev To support non-evm chains, this value is encoded into bytes
+  function getRemoteToken(uint64 remoteChainSelector) public view returns (bytes memory) {
+    return s_remoteChainConfigs[remoteChainSelector].remoteTokenAddress;
+  }
+
+  /// @notice Adds a remote pool for a given chain selector. This could be due to a pool being upgraded on the remote
+  /// chain. We don't simply want to replace the old pool as there could still be valid inflight messages from the old
+  /// pool. This function allows for multiple pools to be added for a single chain selector.
+  /// @param remoteChainSelector The remote chain selector for which the remote pool address is being added.
+  /// @param remotePoolAddress The address of the new remote pool.
+  function addRemotePool(uint64 remoteChainSelector, bytes calldata remotePoolAddress) external onlyOwner {
+    if (!isSupportedChain(remoteChainSelector)) revert NonExistentChain(remoteChainSelector);
+
+    _setRemotePool(remoteChainSelector, remotePoolAddress);
+  }
+
+  /// @notice Removes the remote pool address for a given chain selector.
+  /// @dev All inflight txs from the remote pool will be rejected after it is removed. To ensure no loss of funds, there
+  /// should be no inflight txs from the given pool.
+  function removeRemotePool(uint64 remoteChainSelector, bytes calldata remotePoolAddress) external onlyOwner {
+    if (!isSupportedChain(remoteChainSelector)) revert NonExistentChain(remoteChainSelector);
+
+    if (!s_remoteChainConfigs[remoteChainSelector].remotePools.remove(keccak256(remotePoolAddress))) {
+      revert InvalidRemotePoolForChain(remoteChainSelector, remotePoolAddress);
+    }
+
+    emit RemotePoolRemoved(remoteChainSelector, remotePoolAddress);
+  }
+
+  /// @inheritdoc IPoolV1
+  function isSupportedChain(uint64 remoteChainSelector) public view returns (bool) {
+    return s_remoteChainSelectors.contains(remoteChainSelector);
+  }
+
+  /// @notice Get list of allowed chains
+  /// @return list of chains.
+  function getSupportedChains() public view returns (uint64[] memory) {
+    uint256[] memory uint256ChainSelectors = s_remoteChainSelectors.values();
+    uint64[] memory chainSelectors = new uint64[](uint256ChainSelectors.length);
+    for (uint256 i = 0; i < uint256ChainSelectors.length; ++i) {
+      chainSelectors[i] = uint64(uint256ChainSelectors[i]);
+    }
+
+    return chainSelectors;
+  }
+
+  /// @notice Sets the permissions for a list of chains selectors. Actual senders for these chains
+  /// need to be allowed on the Router to interact with this pool.
+  /// @param remoteChainSelectorsToRemove A list of chain selectors to remove.
+  /// @param chainsToAdd A list of chains and their new permission status & rate limits. Rate limits
+  /// are only used when the chain is being added through `allowed` being true.
```

Comment:
If I understand this comment correctly, the `allowed` bool was removed, so I think this comment is out of date